File: Programmers/CodingTest/hash/problem1.js
// https://school.programmers.co.kr/learn/courses/30/lessons/1845
// 폰켓몬
function solution(nums) {
  const map = new Map();
  const answer = nums.length / 2;

  for (let i = 0; i < nums.length; i++) {
    map.get(nums[i])
      ? map.set(nums[i], map.get(nums[i]) + 1)
      : map.set(nums[i], 1);
  }

  return answer < map.size ? answer : map.size;
}

File: Programmers/CodingTest/hash/problem1.cpp
// https://school.programmers.co.kr/learn/courses/30/lessons/1845
// 폰켓몬
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> nums)
{
    int cnt = nums.size()/2;
    sort(nums.begin(), nums.end());
    nums.erase(unique(nums.begin(), nums.end()), nums.end());
    
    return nums.size() > cnt ? cnt : nums.size();
}